src/mono/wasm/runtime/assets.ts

```diff
@@ -135,28 +134,14 @@ export async function mono_download_assets(): Promise<void> {
     }
 }
 
-export async function start_asset_download(asset: AssetEntryInternal) {
-    // `response.arrayBuffer()` can't be called twice. Some use-cases are calling it on response in the instantiation.
-    const headersOnly = skipBufferByAssetTypes[asset.behavior];
-    if (asset.pendingDownload) {
-        asset.pendingDownloadInternal = asset.pendingDownload;
-        const response = await asset.pendingDownloadInternal!.response;
-        ++actual_downloaded_assets_count;
-        if (!headersOnly) {
-            asset.buffer = await response.arrayBuffer();
-        }
-        return { asset, buffer: asset.buffer };
-    } else {
-        asset.buffer = await start_asset_download_with_retries(asset, !headersOnly);
-        return { asset, buffer: asset.buffer };
-    }
-}
-
 // FIXME: Connection reset is probably the only good one for which we should retry
-async function start_asset_download_with_retries(asset: AssetEntryInternal, downloadData: boolean): Promise<ArrayBuffer | undefined> {
+export async function start_asset_download(asset: AssetEntryInternal): Promise<AssetEntryInternal> {
     try {
-        return await start_asset_download_with_throttle(asset, downloadData);
+        return await start_asset_download_with_throttle(asset);
     } catch (err: any) {
+        if (!runtimeHelpers.enableDownloadRetry) {
+            // we will not re-try if disabled
```

Comment:
```suggestion
            // we will not re-try if disabled
            throw err;
```

We should do something here :)